src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs

```diff
@@ -173,7 +173,7 @@ private IEnumerable<IVisual> HitTest(IVisualNode node, Point p, Rect? clip, Func
                 if (node.GeometryClip != null)
                 {
                     var controlPoint = Root.Visual.TranslatePoint(p, node.Visual);
-                    clipped = !node.GeometryClip.FillContains(controlPoint);
+                    clipped = !node.GeometryClip.FillContains(controlPoint.Value);
```

Comment:
Maybe throw a null reference exception?

Reply:
This is guaranteed to have a value because we know the two controls are in the same tree.